src/sentry/workflow_engine/handlers/condition/event_frequency_base_handler.py

```diff
@@ -166,3 +178,47 @@ def get_rate_bulk(
                 environment_id=environment_id,
             )
         return result
+
+
+class BaseEventFrequencyCountHandler:
+    comparison_json_schema = {
+        "type": "object",
+        "properties": {
+            "interval": {"type": "string", "enum": list(STANDARD_INTERVALS.keys())},
+            "value": {"type": "integer", "minimum": 0},
+        },
+        "required": ["interval", "value"],
+        "additionalProperties": False,
+    }
+
+    @staticmethod
+    def evaluate_value(value: WorkflowJob, comparison: Any) -> DataConditionResult:
+        if len(value.get("snuba_results", [])) != 1:
+            return False
+        return value["snuba_results"][0] > comparison["value"]
+
+
+class BaseEventFrequencyPercentHandler:
+    comparison_json_schema = {
+        "type": "object",
+        "properties": {
+            "interval": {"type": "string", "enum": list(STANDARD_INTERVALS.keys())},
+            "value": {"type": "integer", "minimum": 0},
+            "comparison_interval": {"type": "string", "enum": list(COMPARISON_INTERVALS.keys())},
+        },
+        "required": ["interval", "value", "comparison_interval"],
+        "additionalProperties": False,
+    }
+
+    @property
+    def intervals(self) -> dict[str, tuple[str, timedelta]]:
+        return PERCENT_INTERVALS
+
+    @staticmethod
+    def evaluate_value(value: WorkflowJob, comparison: Any) -> DataConditionResult:
+        if len(value.get("snuba_results", [])) != 2:
+            return False
+        return (
+            percent_increase(value["snuba_results"][0], value["snuba_results"][1])
+            > comparison["value"]
+        )
```

Comment:
these were pulled out from the `EventFrequencyHandlers` since they are shared amongst all count and percent frequency handlers, respectively